src/app.cpp

```diff
@@ -3,44 +3,57 @@
 #include "yaml-cpp/yaml.h"
 
 #include "../include/app.h"
+#include "../include/learning/HeuristicPipeline.h"
+#include "../include/types/AlternativesPerformance.h"
+#include "../include/types/DataGenerator.h"
 
+#include <filesystem>
 #include <iostream>
 
 void App::initializeLogger(YAML::Node &yml_conf) {
+
   std::string log_path;
   if (yml_conf["log_file"]) {
-    log_path = yml_conf["log_file"].as<std::string>();
-  } else {
-    log_path = "../logs/app_log.txt";
+    conf.log_file = yml_conf["log_file"].as<std::string>();
   }
-  conf.logger = spdlog::basic_logger_mt("app_logger", log_path);
+  conf.logger = spdlog::basic_logger_mt("app_logger", conf.log_file);
 
-  if (yml_conf["env"][conf.env]["log_level"].as<std::string>() == "DEBUG") {
+  if (yml_conf["log_level"].as<std::string>() == "DEBUG") {
     spdlog::set_level(spdlog::level::debug);
-  } else if (yml_conf["env"][conf.env]["log_level"].as<std::string>() ==
-             std::string("INFO")) {
+  } else if (yml_conf["log_level"].as<std::string>() == std::string("INFO")) {
     spdlog::set_level(spdlog::level::info);
-  } else if (yml_conf["env"][conf.env]["log_level"].as<std::string>() ==
-             std::string("WARN")) {
+  } else if (yml_conf["log_level"].as<std::string>() == std::string("WARN")) {
     spdlog::set_level(spdlog::level::warn);
-  } else if (yml_conf["env"][conf.env]["log_level"].as<std::string>() ==
-             std::string("ERROR")) {
+  } else if (yml_conf["log_level"].as<std::string>() == std::string("ERROR")) {
     spdlog::set_level(spdlog::level::err);
-  } else if (yml_conf["env"][conf.env]["log_level"].as<std::string>() ==
+  } else if (yml_conf["log_level"].as<std::string>() ==
              std::string("CRITICAL")) {
     spdlog::set_level(spdlog::level::critical);
   } else {
     spdlog::set_level(spdlog::level::info);
   }
+  conf.logger->flush_on(spdlog::level::info);
+}
+
+void App::extractConfig(YAML::Node &yml_conf) {
+  if (yml_conf["data_dir"]) {
+    conf.data_dir = yml_conf["data_dir"].as<std::string>();
+  }
+  if (yml_conf["model_batch_size"]) {
+    conf.model_batch_size = yml_conf["model_batch_size"].as<int>();
+  }
+  if (yml_conf["max_iterations"]) {
+    conf.max_iterations = yml_conf["max_iterations"].as<int>();
+  }
+  this->initializeLogger(yml_conf);
 }
 
 void App::showUsage(std::string name) {
-  std::cerr
-      << "Usage: " << name << " <option(s)> SOURCES"
-      << "Options:\n"
-      << "\t-h,--help\t\tShow this help message\n"
-      << "\t-e,--env ENVIRONMENT\tSpecify the environment: staging or prod"
-      << std::endl;
+  std::cerr << "Usage: " << name << " <option(s)> SOURCES"
+            << "Options:\n"
+            << "\t-h,--help\t\tShow this help message\n"
+            << "\t-d,--dataset DATASET\tDataset path\n"
```

Comment:
maybe add "Dataset FILE path" and "Output FILE path" to ensure they don't enter a directory?